photon-test/src/main/scala/com/linkedin/photon/ml/test/TestTemplateWithTmpDir.scala

```diff
@@ -24,15 +27,25 @@ import org.testng.annotations.AfterMethod
  * @author dpeng
  */
 trait TestTemplateWithTmpDir {
+
+  private val tmpDirs = new mutable.ArrayBuffer[String]()
+
   /**
    * Return the temporary directory as a string.
    * @return the temporary directory as a string.
    */
-  def getTmpDir: String = TestTemplateWithTmpDir._tmpDirThreadLocal.get()
+  def getTmpDir: String = {
+    val tmpDirName = TestTemplateWithTmpDir._tmpDirThreadLocal.get()
+    tmpDirs.synchronized {
+      tmpDirs += tmpDirName
+    }
+    tmpDirName
+  }
 
   @AfterMethod
   def afterMethod(): Unit = {
-    FileUtils.cleanDirectory(new File(getTmpDir))
+    tmpDirs.foreach(tmpDir => FileUtils.cleanDirectory(new File(tmpDir)))
```

Comment:
According to the doc, I think you need this:
https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#deleteDirectory(java.io.File)